### PR TITLE
Malaria-101 Activity-3

### PR DESCRIPTION
**Pull request by Raxerz**
This module contains the code for the third activity under Malaria 101 stage.
It contains a quiz on the prevention of malaria.

Video : https://drive.google.com/file/d/0B7Hnx8cG8DNBWDYxQ0JFdG9fRWs/view?usp=sharing
Responsiveness video : https://drive.google.com/file/d/0B7Hnx8cG8DNBSHA5bEkzMTFrUHc/view?usp=sharing

### CODE REVIEW

**Comment by yukie1412** [comment]
screen shot 2017-08-12 at 10 42 49 pm

text aren't vertically center aligned, there's too much bottom margin/padding in comparison to the top

File: data/quiz.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            "question":"Disease associated with malaria",
+            "option1":"Jaundice",
+            "option2":"Tuberculosis",
+            "option3":"Cholera",
+            "option4":"Typhoid",
+            "answer":"1"
+        }                            
+    ]
+}
```
Comment (yukie1412) [inline data/quiz.json]: newline

File: src/app/malaria-101/activity-3/activity-3.component.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            <div id= "1" class="option" (click)="optionSelected($event)">{{this.opt[0]}}</div>
+            <div id= "2" class="option" (click)="optionSelected($event)">{{this.opt[1]}}</div>
+        </div>
+        <div class="col-lg-4">
+            <div id= "3" class="option" (click)="optionSelected($event)">{{this.opt[2]}}</div> 
+            <div id= "4" class="option" (click)="optionSelected($event)">{{this.opt[3]}}</div>  
+        </div>
+    </div>
+    <div class="row pad" [hidden]="!activityComplete">
+        <div class="swal2-icon swal2-info" style="display: block;">i</div>
```
Comment (yukie1412) [inline src/app/malaria-101/activity-3/activity-3.component.html]: inline styling

File: src/app/malaria-101/activity-3/activity-3.component.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    <br><br>
+    <div class="row">
+        <button prev class="col-lg-5 col-md-offset-3">
+            Prev
+        </button>
+        <button next class="col-lg-5 col-md-offset-6" [hidden]="!showNext">
+            Next
+        </button>
+    </div>
+</div>
```
Comment (yukie1412) [inline src/app/malaria-101/activity-3/activity-3.component.html]: newline

File: src/app/malaria-101/activity-3/activity-3.component.html [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<div class="quiz-container">
+    <div class="row" [hidden]="activityComplete">
+        <h2 class = "col-lg-5 col-lg-offset-4 text-center" >{{questionText}}</h2>
+    </div>
+    <div class="row col-md-offset-4" [hidden]="activityComplete">
+        <br><br>
```
Comment (yukie1412) [inline src/app/malaria-101/activity-3/activity-3.component.html]: not a fan of `<br><br>` for line breaks
this should be done via css

File: src/app/malaria-101/activity-3/activity-3.component.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        this.opt.push(this._data.quizlist[this._questionNumber].option3);
+        this.opt.push(this._data.quizlist[this._questionNumber].option4);
+    }
+
+    /**
+     * Check if the option selected is right or wrong
+     * @param {Object} event Use click event to get the attributes of the current DOM
+     */
+    optionSelected(event) {
+        if (this._questionLock === false) {
```
Comment (yukie1412) [inline src/app/malaria-101/activity-3/activity-3.component.ts]: why not just `if (this._questionLock) {return;}`?

File: src/app/malaria-101/malaria-101.component.scss [hunk 6]
```diff
@@ -0,0 +1,238 @@
+@import "../styles/main";
+
+video {
```
Comment (yukie1412) [inline src/app/malaria-101/malaria-101.component.scss]: believe i've seen styling for `video` elsewhere

File: src/app/malaria-101/malaria-101.component.scss [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+}
+
+@mixin text($topPos, $leftPos) {
+    position: absolute;
+    width: 15%;
+    top: $topPos;
+    left: $leftPos;
+}
+
+@mixin rotate($rotation) {
```
Comment (yukie1412) [inline src/app/malaria-101/malaria-101.component.scss]: don't we have a `rotate` mixin elsewhere?

actually this entire file feels duplicated

File: src/app/malaria-101/malaria-101.component.scss (hunk 6 above)
Comment (Raxerz) [inline src/app/malaria-101/malaria-101.component.scss]: Yes, this is the same file as the one in my other open PR.
It has additional css for the new activity

File: src/app/malaria-101/activity-3/activity-3.component.html [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<div class="quiz-container">
+    <div class="row" [hidden]="activityComplete">
+        <h2 class = "col-lg-5 col-lg-offset-4 text-center" >{{questionText}}</h2>
+    </div>
+    <div class="row col-md-offset-3" [hidden]="activityComplete">
+        <div class="col-lg-5">
+            <div class="option" (click)="optionSelected($event)"><span id= "1">{{this.opt[0]}}</span></div>
```
Comment (yukie1412) [inline src/app/malaria-101/activity-3/activity-3.component.html]: spacing

File: src/app/malaria-101/malaria-101.component.scss [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+
+.quiz-container {
+    margin-top: 7%;
+}
+
+.option {
+    display: table;
+    margin: 8px;
+    padding: 8px;
+    background: #FAE3BB;
```
Comment (yukie1412) [inline src/app/malaria-101/malaria-101.component.scss]: use variables?

**Comment by Raxerz** [comment]
Made the changes @yukie1412 . Can you please check?

File: src/app/malaria-101/activity-3/activity-3.component.ts [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+    /**
+     * Logic to display the next question
+     */
+    changeQuestion() {
+        this._subscription.unsubscribe();
+        this._questionNumber++;
+        if (this._questionNumber === 5) {
+            this.activityComplete = true;
+            this.showNext = true;
```
Comment (yukie1412) [inline src/app/malaria-101/activity-3/activity-3.component.ts]: is this needed? you aren't setting it to false nor using it anywhere

**Comment by Raxerz** [comment]
I made the modification @yukie1412, I am using it in the html file now.

**Comment by Raxerz** [comment]
Since this value is required only once, there is no need to reset it

**Pull request merged**